### DISCUSSION TITLE
Fregate and Coral dual TPU settings stopped working!?

### DISCUSSION THREAD

**Discussion by Trzinka** [blakeblackshear/frigate#15713] [Ask A Question]
Hi!

I run Home Assistant on Proxmox (all latest versions "updated")

In Proxmox I see:
```yaml
08:00.0 System peripheral: Global Unichip Corp. Coral Edge TPU
09:00.0 System peripheral: Global Unichip Corp. Coral Edge TPU
```

In the home assistant terminal I get the following:
![Frigat-Coral TPU](https://github.com/user-attachments/assets/19134ae5-1679-4084-8132-2779deea8dee)

My settings for coral duo TPU in frigate.yml:
```yaml
detectors:
  coral1:
    type: edgetpu
    device: pci:0
  coral2:
    type: edgetpu
    device: pci:1
```

In the log I can see:
```yaml
2024-12-25 10:19:25.022360396  [2024-12-25 11:19:25] frigate.detectors.plugins.edgetpu_tfl ERROR   : No EdgeTPU was detected. If you do not have a Coral device yet, you must configure CPU detectors.
2024-12-25 10:19:25.023397485  Process detector:coral1:
2024-12-25 10:19:25.024410941  Traceback (most recent call last):
2024-12-25 10:19:25.024661314    File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 160, in load_delegate
2024-12-25 10:19:25.024663242      delegate = Delegate(library, options)
2024-12-25 10:19:25.024664369    File "/usr/lib/python3/dist-packages/tflite_runtime/interpreter.py", line 119, in __init__
2024-12-25 10:19:25.024669955      raise ValueError(capture.message)
2024-12-25 10:19:25.024670842  ValueError
```

`I've been using the M.2 Accelerator with Dual Edge TPU on an 8 core PC with 32 GB RAM for almost 2 years without any major problems!`

If I run Frigate with these settings it works for me (of course worse):
```yaml
detectors:
  cpu1:
    type: cpu
  cpu2:
    type: cpu   
#  coral1:
#    type: edgetpu
#    device: pci:0
#  coral2:
#    type: edgetpu
#    device: pci:1
```

**Any idea what could be wrong?**

Thanks!
Happy holidays and a happy new year.

**Comment by hawkeye217** [comment]
Did you try a good old fashioned hard reboot of the entire system?

**Reply by Trzinka** [reply]
Hi. Thanks for the answer.
Yes. Stopped and restarted Proxmox and everything running on it..

**Reply by Trzinka** [reply]
`Is this the correct display of Coral TPU within Home Assistant?`
![HA-Apex-Coral TPU](https://github.com/user-attachments/assets/c4504f6f-ddf2-4bd0-9423-fc44da0b525a)

**Reply by hawkeye217** [reply]
Right, but did you do a hard reboot? As in, did you remove power from the entire system and restart it?

**Reply by Trzinka** [reply]
I waited 6 hours before restarting the computer (power went out) ;=((((

**Reply by Trzinka** [reply]
I even run frigates with (Full Access) and disabling protection mode!

**Reply by Trzinka** [reply]
![Frigate Full acces](https://github.com/user-attachments/assets/9e9a4482-c87a-470d-9417-2c26ce94407c)

**Reply by hawkeye217** [reply]
Did something update when you rebooted? Proxmox is technically an unsupported installation method, so maybe some other users can help.

**Reply by Trzinka** [reply]
I'm not exactly a rocket scientist, but I've been working with computers for 35 years so I know some basics. The problem is that I don't have enough knowledge about the Linux system.

`As I mentioned, something apparently happened after upgrading Home Assistant. I read somewhere that the problem could be incompatibility between Python versions. But I don't have enough knowledge to say that.` But Thanks.

**Reply by Cold-Lemonade** [reply]
@Trzinka Have you had the dual coral working before in Proxmox?

It is not enough that you see the devices when you run ```lspci```. You need to install the apex driver for it. You should see them listed as ```/dev/apex0``` and ```/dev/apex1``` or something like that in the ```/dev``` directory. I run Frigate in an LXC container in Proxmox and you'll need to make sure you pass those devices to the LXC container. There's lots of discussion about how to install the apex drivers on here and the Proxmox forum. Please note that Google's instructions for how to install it are not correct as one of the dependencies needs to be built manually. Good luck!

**Reply by Trzinka** [reply]
What does the following error mean:
```yaml
2025-01-04 07:40:34.319679599 /usr/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 16 leaked shared_memory objects to clean up at shutdown
2025-01-04 07:40:34.319682452 warnings.warn('resource_tracker: There appear to be %d '
2025-01-04 07:40:34.322853785 [INFO] Service Frigate exited with code 0 (by signal 0)```?

**Reply by Trzinka** [reply]
When I restart my computer, Frigate works for a while (I can't find a logical time interval) and then it stops working with the aforementioned errors.

**Reply by Trzinka** [reply]
This error is being reported to me in the home assistant terminal
```yaml
apex 0000:06:11.0: RAM did not enable within timeout (12000 ms)
apex 0000:06:11.0: Error in device open cb: -110

```
